### PR TITLE
Porting jsf-2.2 FAT Tests

### PR DESCRIPTION
**Pull request by volosied**
For issue #7526

All tests pass successfully.  Also includes jsf.2.3 repeat tests. 

All test classes were moved to OL  except for the two below
                 * JSF22JSF20SingletonFeatureTest.class - moved to com.ibm.ws.jsf_fat
                 * JSF22JPA20Test.class  - moved to  com.ibm.ws.jpa_fat

### CODE REVIEW

**Comment by volosied** [comment]
#build #build.fullfat.buckets=com.ibm.ws.jsf.2.2_fat

File: dev/com.ibm.ws.jsf.2.2_fat/.classpath [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+<classpathentry kind="src" output="bin/main" path="test-applications/PI85492.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/PI90507.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/JSF22BackwardCompatibilityTests.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/JSF22HTML5.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/JSF22SimpleHTML.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/CDICommon/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/JSF22InputFile.war/src">
+<classpathentry kind="src" output="bin/main" path="test-applications/PH01566.war/src">
+<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/.classpath]: I'm not familiar with the gradleclasspathcontainer, is this used in other test projects? Usually I see `<classpathentry kind="con" path="aQute.bnd.classpath.container"/>`

File: dev/com.ibm.ws.jsf.2.2_fat/.project [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		<buildCommand>
+			<name>bndtools.core.bndbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/.project]: Just want you to verify this is necessary `gradleprojectnature`

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/FATSuite.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+ * use lite mode (<code>@Mode(TestMode.LITE)</code>).
+ * 
+ * By default only lite mode tests are run. To also run
+ * full mode tests a property must be specified to ant:
+ * 
+ * Select the target build file (usually "build-test.xml").
+ * Right click and chose "Run As>Ant Build…". Add
+ * "fat.test.mode=full" to the properties tab, then launch the
+ * build.
+ *
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/FATSuite.java]: Comment here can be cleaned up as we don't use ANT anymore for instance.

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/CDIConfigByACPTests.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+ * CDIConfigByACP - Application Configuration Populator loading of the class files
+ *
+ * We're extending CDITestBase, which has common test code which is found under test-applications/CDICommon.
+ *
+ * NOTE: These tests should not run with jsf-2.3 feature because constructor injection is not supported.
+ * As a result, these tests were modified to run in the JSF 2.3 FAT bucket without constructor injection.
+ */
+@Mode(TestMode.FULL)
+@SkipForRepeat("JSF-2.3")
+@MinimumJavaLevel(javaLevel = 7)
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/CDIConfigByACPTests.java]: To the best of my knowledge we don't need to specify the MinimumJavaLevel at the class level if you're already specifying it in the bnd.bnd : https://github.com/OpenLiberty/open-liberty/wiki/FAT-tests#running-fat-tests-locally-with-different-java-levels

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/CDIConfigByACPTests.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+ * This is one of four CDI test applications, with configuration loaded in the following manner:
+ * CDIConfigByACP - Application Configuration Populator loading of the class files
+ *
+ * We're extending CDITestBase, which has common test code which is found under test-applications/CDICommon.
+ *
+ * NOTE: These tests should not run with jsf-2.3 feature because constructor injection is not supported.
+ * As a result, these tests were modified to run in the JSF 2.3 FAT bucket without constructor injection.
+ */
+@Mode(TestMode.FULL)
+@SkipForRepeat("JSF-2.3")
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/CDIConfigByACPTests.java]: Why don't we want to run these tests with JSF -2.3?

File: dev/com.ibm.ws.jsf.2.3_fat/.project [hunk 6]
```diff
@@ -1,23 +1,29 @@
...
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/.project]: Can you give me some information on the updates here for JSF 2.3 in this PR?

**Comment by volosied** [comment]
Build passes except for 3 unrelated test failures:

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_pvlB0LreEempGelF87C7VA&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/FATSuite.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                JSF22InputFileTests.class,
+                JSF22LocalizationTesterTests.class,
+                JSF22AparTests.class
+
+                /*
+                 * Tests below were in the 2.2 CL Fat Bucket, but could not be ported to OL.
+                 * 
+                 * JSF22JSF20SingletonFeatureTest.class -- moved to com.ibm.ws.jsf_fat
+                 * JSF22JPA20Test.class -- moved to com.ibm.ws.jpa_fat
+                 */
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/FATSuite.java]: Lets just remove this comment.

Comment (volosied) [reply dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/FATSuite.java]: removed

File: dev/com.ibm.ws.jsf.2.2_fat/bnd.bnd [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    test-applications/PI67525.war/src,\
+    test-applications/PI90507.war/src,\
+    test-applications/PH01566.war/src,\
+    test-applications/JSF22LocalizationTester.war/src
+    
+fat.project: true
+tested.features: servlet-4.0, jsf-2.3, cdi-2.0, beanValidation-2.0
+
+runtime.minimum.java.level: 1.7
+
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.2_fat/bnd.bnd]: The `runtime.minimum.java.level: 1.7` is not necessary given the javac.source/target defs below.

**Comment by pnicolucci** [comment]
Lets make the last code review changes and then get another build and ensure to run the bucket in full mode as I want to ensure the RepeatTests is working correctly also.

File: dev/com.ibm.ws.jsf.2.2_fat/bnd.bnd (hunk 8 above)
Comment (volosied) [reply dev/com.ibm.ws.jsf.2.2_fat/bnd.bnd]: removed

**Pull request merged**